tests/components/datadog/test_init.py

```diff
@@ -145,3 +166,97 @@ async def test_state_changed(hass: HomeAssistant) -> None:
             hass.states.async_set("domain.test", invalid, {})
             await hass.async_block_till_done()
             assert not mock_statsd.gauge.called
+
+
+async def test_unload_entry(hass: HomeAssistant) -> None:
+    """Test unloading the config entry cleans up properly."""
+    client = mock.MagicMock()
+
+    with (
+        patch("homeassistant.components.datadog.DogStatsd", return_value=client),
+        patch("homeassistant.components.datadog.initialize"),
+    ):
+        entry = MockConfigEntry(
+            domain=datadog.DOMAIN,
+            data=MOCK_DATA,
+            options=MOCK_OPTIONS,
+        )
+        entry.add_to_hass(hass)
+
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert await hass.config_entries.async_unload(entry.entry_id)
+        await hass.async_block_till_done()
+
+    client.flush.assert_called_once()
+    client.close_socket.assert_called_once()
+
+
+async def test_state_changed_skips_unknown(hass: HomeAssistant) -> None:
+    """Test state_changed_listener skips None and unknown states."""
+    entry = MockConfigEntry(domain=datadog.DOMAIN, data=MOCK_DATA, options=MOCK_OPTIONS)
+    entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.components.datadog.config_flow.DogStatsd"
+        ) as mock_dogstatsd,
+    ):
+        await async_setup_entry(hass, entry)
+
+        # Test None state
+        hass.bus.async_fire(EVENT_STATE_CHANGED, {"new_state": None})
+        await hass.async_block_till_done()
+        assert not mock_dogstatsd.gauge.called
+
+        # Test STATE_UNKNOWN
+        unknown_state = mock.MagicMock()
+        unknown_state.state = STATE_UNKNOWN
+        hass.bus.async_fire(EVENT_STATE_CHANGED, {"new_state": unknown_state})
+        await hass.async_block_till_done()
+        assert not mock_dogstatsd.gauge.called
+
+
+async def test_yaml_import_creates_deprecation_issue(hass: HomeAssistant) -> None:
```

Comment:
We already have these test cases in `test_config_flow.py` so extend it there to assert issues has been created. No need to have separate test cases.